pkg/tecdsa/dkg/dkg.go

```diff
@@ -95,6 +95,74 @@ func (e *Executor) Execute(
 	return finalizationState.result(), endBlockNumber, nil
 }
 
+// ResultSigner is the interface that provides ability to sign the DKG result
+// and verify the results received from other group members.
+type ResultSigner interface {
+	// SignResult signs the provided DKG result. It returns the public key used
+	// during the signing, the resulting signature and the result hash.
+	SignResult(result *Result) ([]byte, []byte, ResultHash, error)
+	// VerifySignature verifies if the signature was generated from the provided
+	// message using the provided public key.
+	VerifySignature(message []byte, signature []byte, publicKey []byte) (bool, error)
+}
+
+// ResultSubmitter is the interface that provides ability to submit the DKG
+// result to the chain.
+type ResultSubmitter interface {
+	SubmitResult(
+		result *Result,
+		signatures map[group.MemberIndex][]byte,
+		startBlockNumber uint64,
+		participantIndex group.MemberIndex,
+	) error
+}
+
+// Publish signs the DKG result for the given group member, collects signatures
+// from other members and verifies them, and submits the DKG result.
+func Publish(
+	logger log.StandardLogger,
+	seed *big.Int,
```

Comment:
Let's pass the `sessionID string` directly (see #3169)

Reply:
Done in 62983725edba3fe648e1dda23d6fe01e831799cc.